R/modules.R

```diff
@@ -321,11 +321,124 @@ modules <- function(..., label = "root") {
 # printing methods ----
 
 #' @rdname teal_modules
+#' @param what character vector specifying which metadata to display. 
+#'   Possible values: "data_sources", "properties", "ui_args", "server_args", "transformers"
 #' @export
-format.teal_module <- function(x, indent = 0, ...) {
-  paste0(paste(rep(" ", indent), collapse = ""), "+ ", x$label, "\n", collapse = "")
+format.teal_module <- function(x, indent = 0, is_last = FALSE, parent_prefix = "", what = c("data_sources", "properties", "ui_args", "server_args", "transformers"), ...) {
+  empty_text <- "─ none"
+  branch <- if (is_last) "└─" else "├─"
+  current_prefix <- paste0(parent_prefix, branch, " ")
+  content_prefix <- paste0(parent_prefix, if (is_last) "   " else "│  ")
+
+  format_list <- function(lst, empty = empty_text) {
+    if (is.null(lst) || length(lst) == 0) {
+      empty
+    } else {
+      paste(names(lst), collapse = ", ")
+    }
+  }
+
+  bookmarkable <- ifelse(isTRUE(attr(x, "teal_bookmarkable")), "Yes", "No")
+  reportable <- ifelse("reporter" %in% names(formals(x$server)), "Yes", "No")
+
+  transformers <- if (length(x$transformers) > 0) {
+    paste(sapply(x$transformers, function(t) attr(t, "label")), collapse = ", ")
+  } else {
+    empty_text
+  }
+
+  output <- paste0(current_prefix, crayon::bold(x$label), "\n")
+  
+  if ("data_sources" %in% what) {
+    output <- paste0(
+      output,
+      content_prefix, "├─ ", crayon::blue("Data Sources:"), " ", paste(x$datanames, collapse = ", "), "\n"
+    )
+  }
+  if ("properties" %in% what) {
+    output <- paste0(
```

Comment:
Can you keep `output` as a list, so you don't need to keep pasting. And then at the end of the function, can you unlist everything and paste once?

Reply:
I think this is how you handle this in format.teal_modules

Reply:
When I wend with the implementation I **_feel_** that this makes it complex because the `paste0` is not avoided here.
We will have to use both the paste and appending to the list together, now we just do one.
For the modules it was different, we were fetching and storing it.

Reply:
It's creating a list at the begninning
`list_with_lines <- list()`
and then changing paste into
```
new_line <- pasting stuff
list_with_lines <- append(list_with_lines, new_line)
```

This way you can work on a whole output to adjust the width.
However you can now also work on the output to adjust the width. You can split by `\n` sign so the whole thing with keeping output in a list is not a big deal. 

Just trying to sync our print/format method with the way it was written for teal.slice https://github.com/insightsengineering/teal.slice/blob/main/R/teal_slice.R

Reply:
I think that's how we can control the width.

```r
# complete_modules is taken from your example from the opening of the discussion on this PR
cat(
  substr(capture.output(complete_modules), 1, 40),
  sep = "\n"
)
```